packaging/bin/create-deb-repo.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+set -eEuo pipefail
+
+# External dependencies:
+# - https://salsa.debian.org/apt-team/apt (apt-ftparchive, packaged in apt-utils)
+# - https://github.com/s3tools/s3cmd
+#   s3cmd expects AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY to be set in the
+#   environment.
+# - https://gnupg.org/
+#   For signing the script expects the private signing key to already be
+#   imported and PGPKEYID and PGP_SIGN_KEY_PASSPHRASE to be set in the
+#   environment.
+# - generate_index.py
+#   For generating the index.html of each directory. It's available in the
+#   packaging/bin directory of the k6 repo, and should be in $PATH.
+
+_s3bucket="${S3_BUCKET-dl.k6.io}"
+_usage="Usage: $0 <pkgdir> <repodir> [s3bucket=${_s3bucket}]"
+PKGDIR="${1?${_usage}}"  # The directory where .deb files are located
+REPODIR="${2?${_usage}}" # The package repository working directory
+S3PATH="${3-${_s3bucket}}/deb"
+# Number of packages to keep for each architecture, older packages will be deleted.
+RETAIN_PKG_COUNT=25
+
+log() {
+    echo "$(date -Iseconds) $*"
+}
+
+delete_old_pkgs() {
+  find "$1" -name '*.deb' -type f | sort -r | tail -n "+$((RETAIN_PKG_COUNT+1))" | xargs -r rm -v
+  # Remove any dangling .asc files
+  find "$1" -name '*.asc' -type f -print0 | while read -r -d $'\0' f; do
+    if ! [ -r "${f%.*}" ]; then
+      rm -v "$f"
+    fi
+  done
+}
+
+sync_to_s3() {
+  log "Syncing to S3 ..."
+  s3cmd sync --delete-removed "${REPODIR}/" "s3://${S3PATH}/"
+
+  # Disable cache for repo metadata and index files, so that new releases will be
+  # available immediately.
+  s3cmd modify --add-header="Cache-Control:no-cache, max-age=0" \
+    "s3://${S3PATH}/dists/stable/"{Release,Release.gpg,InRelease}
+  s3cmd modify --add-header="Cache-Control:no-cache, max-age=0" \
+    "s3://${S3PATH}/dists/stable/main/binary-amd64"/Packages{,.gz,.bz2}
+  s3cmd modify --recursive --exclude='*' --include='index.html' \
+    --add-header='Cache-Control:no-cache, max-age=0' "s3://${S3PATH}/"
```

Comment:
So this is not that simple :( Passing `--cf-invalidate` instead actually does nothing and I still get hits from CF. It seems this option requires some additional IAM permissions (`cloudfront:CreateInvalidation` at the very least), but there's no s3cmd documentation about it, I can't see anything useful with `-v --debug` (I would at least expect it to fail...), and I can't find anything in the repo issues, except old issues claiming that it doesn't work (536, 790).

s3cmd seemed like the simplest tool to use that's still decently maintained, but I could switch to the official `aws-cli` for this which has a standalone `aws cloudfront` command. For now I'll keep the `Cache-Control` header, even though it means these files will never be cached, which should be fine since they're small index and metadata files.

Reply:
:sob:

Reply:
I think we can rethink,refactor this after a month or two if it turns out that it is a problem or is costing too much money so ... just drop it for now

Reply:
So, good news... I switched to awscli instead of s3cmd, and CF invalidation works with `aws cloudfront`. I still have some work to do to add the permission in the IAM policy with Terraform (currently I just added it manually), and to test the latest MSI with S3 redirects again since I wasn't able to get it working with `s3cmd` and ended up duplicating the file, but this seems much better than setting `no-cache` for these files. I'll push the changes here once I test them thoroughly in my fork.

Thanks for the suggestion!

Reply:
See the latest changes with the awscli switch. We still need to set a short 60s `max-age` for the index files to avoid browsers caching them, and I did the same for the repo metadata files just in case. But now they're properly invalidated in CloudFront.

Also I couldn't get the S3 redirect to work for the latest MSI package, as it would always be cached by CloudFront for some reason. So I settled on just copying the actual latest file instead.